src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/TypePreinit.cs

```diff
@@ -1415,6 +1415,27 @@ private Status TryScanMethod(MethodIL methodIL, Value[] parameters, Stack<Method
                         }
                         break;
 
+                    case ILOpcode.ldind_i4:
+                    case ILOpcode.ldind_u4:
+                        {
+                            StackEntry entry = stack.Pop();
+                            if (entry.Value is ByRefValue byRefVal)
+                            {
+                                switch (opcode)
+                                {
+                                    case ILOpcode.ldind_i4:
```

Comment:
Are 32-bit ints the only case that shows up in CoreLib today?

Should we handle other bit lengths here as well for completeness?

Reply:
There are definitely longs too e.g. in `FormattingHelpers.CountDigits.cs`

Reply:
These are our current uses:
<img width="510" alt="image" src="https://user-images.githubusercontent.com/2642209/217342017-093ac879-5728-4d57-8e79-44b6ab1e0c91.png">

Reply:
> Should we handle other bit lengths here as well for completeness?

The IL interpreter doesn't aim for completeness. I implemented the ones that showed up as blockers for the app I was looking at. E.g. the `CalendricalCalculationHelpers` with the float use case are blocked on some other unimplemented thing in the interpreter, so it doesn't show up.

Others are welcome to extend the interpreter and write the necessary tests.

Reply:
I agree that the IL interpreter does not aim for completeness, but I view this as lame incompleteness.

100% test coverage would be nice, but I do not think it is strictly required for repetitive code. The current change does not seem to have full test coverage either (the tests cover signed int only).

Reply:
The reason why I'd want test coverage is because int/uint don't have interesting behaviors around widening (there isn't any), but float/sbyte/etc. do and those generally make me more anxious.

One could say that not supporting `ReadOnlySpan<>.Length` is also a lame incompleteness. I almost added that too but I didn't run into it and decided it's not worth the trouble.

The CI run seems to indicate I still managed to mess up something so I'll have to look at all this in more detail.